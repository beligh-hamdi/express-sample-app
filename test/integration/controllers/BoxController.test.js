const app = require('../../../api/server');
const request = require('supertest')(app);
const db = require('../../db');

beforeEach(() => db.clear().then(() => db.init()));

describe(':create', () => {
  test('Should create box', (done) => {
    request
    .post('/api/boxes')
    .set('Authorization', 'token')
    .send({ name: 'box' })
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      delete res.body._id;
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should get error (no box name)', (done) => {
    request
    .post('/api/boxes')
    .set('Authorization', 'token')
    .expect(400)
    .end((err, res) => {
      if (err) fail(err);
      delete res.body._id;
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});

describe(':read', () => {
  test('Should get boxes', (done) => {
    request
    .get('/api/boxes')
    .set('Authorization', 'token')
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});

describe(':readOne', () => {
  test('Should get box', (done) => {
    request
    .get('/api/boxes/589d95ea18d583282623b042')
    .set('Authorization', 'token')
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should get no box', (done) => {
    request
    .get('/api/boxes/589d95ea18')
    .set('Authorization', 'token')
    .expect(404)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});

describe(':update', () => {
  test('Should update box', (done) => {
    request
    .put('/api/boxes/589d95ea18d583282623b042')
    .set('Authorization', 'token')
    .send({ name: 'newName' })
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should error on update box. (bad box id)', (done) => {
    request
    .put('/api/boxes/14214')
    .set('Authorization', 'token')
    .send({ name: 'newName' })
    .expect(500)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});

describe(':delete', () => {
  test('Should delete box', (done) => {
    request
    .delete('/api/boxes/589d95ea18d583282623b042')
    .set('Authorization', 'token')
    .expect(200)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });

  test('Should error on delete box. (bad box id)', (done) => {
    request
    .delete('/api/boxes/14214')
    .set('Authorization', 'token')
    .send({ name: 'newName' })
    .expect(400)
    .end((err, res) => {
      if (err) fail(err);
      expect(res.body).toMatchSnapshot();
      done();
    });
  });
});
