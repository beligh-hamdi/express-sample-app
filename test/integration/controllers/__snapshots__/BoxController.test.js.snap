exports[`:create Should create box 1`] = `
Object {
  "__v": 0,
  "name": "box",
}
`;

exports[`:create Should get error (no box name) 1`] = `
Object {
  "errors": Array [
    Object {
      "errors": Object {
        "name": Object {
          "kind": "required",
          "message": "Box name required",
          "name": "ValidatorError",
          "path": "name",
          "properties": Object {
            "message": "Box name required",
            "path": "name",
            "type": "required",
          },
        },
      },
      "message": "Box validation failed",
      "name": "ValidationError",
    },
  ],
}
`;

exports[`:delete Should delete box 1`] = `
Object {
  "n": 1,
  "ok": 1,
}
`;

exports[`:delete Should error on delete box. (bad box id) 1`] = `
Object {
  "errors": Array [
    Object {
      "kind": "ObjectId",
      "message": "Cast to ObjectId failed for value \"14214\" at path \"_id\" for model \"Box\"",
      "name": "CastError",
      "path": "_id",
      "stringValue": "\"14214\"",
      "value": "14214",
    },
  ],
}
`;

exports[`:read Should get boxes 1`] = `
Array [
  Object {
    "__v": 0,
    "_id": "589d95ea18d583282623b042",
    "name": "box1",
  },
]
`;

exports[`:readOne Should get box 1`] = `
Object {
  "__v": 0,
  "_id": "589d95ea18d583282623b042",
  "name": "box1",
}
`;

exports[`:readOne Should get no box 1`] = `
Object {
  "errors": Array [
    Object {
      "kind": "ObjectId",
      "message": "Cast to ObjectId failed for value \"589d95ea18\" at path \"_id\" for model \"Box\"",
      "name": "CastError",
      "path": "_id",
      "stringValue": "\"589d95ea18\"",
      "value": "589d95ea18",
    },
  ],
}
`;

exports[`:update Should error on update box. (bad box id) 1`] = `
Object {
  "errors": Array [
    Object {
      "kind": "ObjectId",
      "message": "Cast to ObjectId failed for value \"14214\" at path \"_id\" for model \"Box\"",
      "name": "CastError",
      "path": "_id",
      "stringValue": "\"14214\"",
      "value": "14214",
    },
  ],
}
`;

exports[`:update Should update box 1`] = `
Object {
  "n": 1,
  "nModified": 1,
  "ok": 1,
}
`;
